File: src/mupyjs/run_tests.py
import glob
import sys
from mupyjs.main import prettier
from mupyjs.pass1 import pass1
from mupyjs.parser import parse
from mupyjs.compiler import compile
from mupyjs.AST import pp
from mupyjs.main import run_markdown_tests

def write_file(name, src):
    file = open(name, 'w')
    file.write(src)
    file.close()

def run_test(test_file):
    file = open(test_file, 'r')
    src = file.read()
    file.close()

    prefix = test_file[:-3]
    ast = parse(src)
    write_file(prefix + '.ast0', pp(ast))
    ast = pass1(ast)
    write_file(prefix + '.ast1', pp(ast))
    js = compile(ast)
    write_file(prefix + '.unformatted.js', js)
    write_file(prefix + '.js', prettier(js))

if __name__ == "__main__":
    run_markdown_tests("README.md")
    test_files = glob.glob('./tests/**/*.py', recursive=True)
    #test_files = reversed(test_files)
    print(test_files)
    for test_file in test_files:
        try:
            run_test(test_file)
        except Exception as e:
            print(e);
            print(f"Error running test {test_file}: {e}")

File: js/mupyjs/run_tests.js
import * as runtime from '@/mupyjs/runtime.js';
Object.assign(self, runtime);
import * as glob from '@/glob';
import * as sys from '@/sys';
import { prettier } from '@/mupyjs/main';
import { pass1 } from '@/mupyjs/pass1';
import { parse } from '@/mupyjs/parser';
import { compile_import, compile } from '@/mupyjs/compiler';
import { pp } from '@/mupyjs/AST';
import { run_markdown_tests } from '@/mupyjs/main';
function write_file(name, src) {
  var file = open(name, 'w');
  file.write(src);
  file.close();
}
function run_test(test_file) {
  var file = open(test_file, 'r');
  var src = file.read();
  file.close();
  var prefix = test_file.__getitem__(
    slice(undefined, (3).__neg__(), undefined)
  );
  var ast = parse(src);
  write_file(prefix.__add__('.ast0'), pp(ast));
  ast = pass1(ast);
  write_file(prefix.__add__('.ast1'), pp(ast));
  var js = compile_import(ast);
  write_file(prefix.__add__('.unformatted.js'), js);
  write_file(prefix.__add__('.js'), prettier(js));
}
if ((__name__ ?? Nil).__eq__('__main__')) {
  run_markdown_tests('README.md');
  var test_files = glob.glob('./tests/**/*.py', {
    _kwargs: true,
    recursive: true,
  });
  print(test_files);
  for (var test_file of test_files) {
    try {
      run_test(test_file);
    } catch (e) {
      if (e instanceof Exception) {
        print(e);
        print('Error running test ' + test_file + ': ' + e);
      }
    }
  }
}
